Tools/MiniBrowser/wpe/main.cpp

```diff
@@ -237,6 +237,11 @@ static gboolean wpeViewEventCallback(WPEView* view, WPEEvent* event, WebKitWebVi
         }
     }
 
+    if (keyval == WPE_KEY_Escape) {
+        webkit_web_view_end_immersive_session(webView);
+        return TRUE;
+    }
```

Comment:
Because here we are claiming that we handle ESC even when there's no session active. If we add the property, we could leave the function as void and require to check the property first.